Data.Yaml.decode is deprecated

```
snack-lib> stack ghci YamlToJson.hs

Warning: Couldn't find a component for file target /.../snack/snack-lib/YamlToJson.hs.
This means that the correct ghc options might not be used.
         Attempting to load the file anyway.
Configuring GHCi with the following packages:
GHCi, version 8.4.3: http://www.haskell.org/ghc/  :? for help
[1 of 1] Compiling Main             ( /.../snack/snack-lib/YamlToJson.hs, interpreted )

/.../snack/snack-lib/YamlToJson.hs:16:22: warning: [-Wdeprecations]
    In the use of ‘decode’ (imported from Data.Yaml):
    Deprecated: "Please use decodeEither or decodeThrow,
which provide information on how the decode failed"
   |
16 |     let Just value = Yaml.decode yaml :: Maybe Aeson.Value
   |                      ^^^^^^^^^^^
Ok, one module loaded.
*Main>
```

--- a/snack-lib/YamlToJson.hs
+++ b/snack-lib/YamlToJson.hs
@@ -13,5 +13,5 @@ main :: IO ()
 main = do
     [file] <- getArgs
     yaml <- BS8.readFile file
-    let Just value = Yaml.decode yaml :: Maybe Aeson.Value
+    let Right value = Yaml.decodeEither' yaml :: Either Yaml.ParseException Aeson.Value
     BL8.putStrLn $ Aeson.encode value

